### ISSUE TITLE
ENH: provide as convenience option to save to zip

### ISSUE THREAD

**Issue by martinfleis** [geopandas/geopandas#1587]
xref https://github.com/geopandas/geopandas/issues/1547#issuecomment-679861409

> Is this something we want to provide as convenience option in to_file?
> I regularly find myself creating a zip file in the file browser of the written shapefiles, and an option to do this automatically might be nice.

We could add an option to `to_file` to save the result as an archive (zip). In #1535 we infer if the file we read is zip, so we could even infer if the saved file should be zipped based on suffix.

**Comment by snowman2** [comment]
Potential alternative in #1607.

**Comment by martinfleis** [comment]
@snowman2 inferring zip schema could be useful for GPKG as well, it is often way smaller when zipped. Changing default to GPKG is not necessarily alternative, both issues are independetn I'd say.

**Comment by jorisvandenbossche** [comment]
Yeah, even if we would switch the default driver, people will still be writing shapefiles, so adding the ability to directly save it as a zip will be useful anyway.

**Comment by snowman2** [comment]
Makes sense 👍
